allennlp/models/semantic_parsing/nlvr/nlvr_decoder_step.py

```diff
@@ -132,10 +127,12 @@ def take_step(self,  # type: ignore
         # action_mask: (group_size, num_embedded_actions)
         action_embeddings, embedded_action_mask = self._get_action_embeddings(state,
                                                                               global_actions_to_embed)
-        action_query = self._get_action_query(state, hidden_state, attended_sentence)
+        action_query = torch.cat([hidden_state, attended_sentence], dim=-1)
         # (group_size, action_embedding_dim)
         predicted_action_embedding = self._output_projection_layer(action_query)
```

Comment:
Not related to this PR, but you probably want a non-linearity here.  See #1150, where I added this to the wikitables model.  There's another spot where you probably don't have one but probably want one, too.

Reply:
And, now that I think about it, because this is getting a dot product with action embeddings, I wonder if relu isn't the best non-linearity.  We're basically saying that any embedding dimension with a negative value is entirely ignored, and thus artificially constraining the space that's available to the dot product...  Maybe tanh would be better?

Reply:
Did you find that adding relus worked better?

Reply:
Adding a non-linearity at the decoder input as well, and made both of them tanh.

Reply:
I ran it with tanh instead of relu last night, and it didn't really change anything.  It looked like it was learning faster - epoch 1 performance was a bit higher - but in the end, final performance was within the normal variance that I've seen across runs.  So, it probably doesn't matter.